### PR TITLE
Add support for 846 (5010 version)

### PR DESCRIPTION
**Pull request by dominiqueplante**

### CODE REVIEW

File: src/Machete.X12Schema/V5010/Layouts/LoopQTY_846.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        Segment<QTY> QuantityInformation { get; }
+        
+        Segment<UIT> UnitDetail { get; }
+        
+        Segment<MEA> Measurements { get; }
+        
+        Segment<LDT> LeadTime { get; }
+        
+        Segment<DTM> DateTimeReference { get; }
```
Comment (ahives) [inline src/Machete.X12Schema/V5010/Layouts/LoopQTY_846.cs]: Should be DateOrTimeReference

File: src/Machete.X12Schema/V5010/Layouts/LoopREF_846.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+{
+    using X12;
+
+
+    public interface LoopREF_846 :
+        X12Layout
+    {
+        Segment<REF> ReferenceInformation { get; }
+        
+        Segment<DTM> DateTimeReference { get; }
```
Comment (ahives) [inline src/Machete.X12Schema/V5010/Layouts/LoopREF_846.cs]: Should be DateOrTimeReference

File: src/Machete.X12Schema/V5010/Layouts/LoopSCH_846.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+namespace Machete.X12Schema.V5010
+{
+    using X12;
+
+
+    public interface LoopSCH_846 :
+        X12Layout
+    {
+        Segment<SCH> LineItemSchedule { get; }
+        Segment<MEA> Measurements { get; }
```
Comment (ahives) [inline src/Machete.X12Schema/V5010/Layouts/LoopSCH_846.cs:10]: newline

File: src/Machete.X12Schema/V5010/Layouts/LoopMAN_846.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+namespace Machete.X12Schema.V5010
+{
+    using X12;
+
+
+    public interface LoopMAN_846 : X12Layout
+    {
+        Segment<MAN> MarksAndNumberInformation { get; }
+        Segment<MEA> Measurements { get; }
```
Comment (ahives) [inline src/Machete.X12Schema/V5010/Layouts/LoopMAN_846.cs:10]: newline

File: src/Machete.X12Schema/V5010/Layouts/LoopMAN_846.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+namespace Machete.X12Schema.V5010
+{
+    using X12;
+
+
+    public interface LoopMAN_846 : X12Layout
```
Comment (ahives) [inline src/Machete.X12Schema/V5010/Layouts/LoopMAN_846.cs]: X12Layout should be on new line

File: src/Machete.X12Schema/V5010/Layouts/LoopSLN_846.cs [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    public interface LoopSLN_846 : 
+        X12Layout
+    {
+        Segment<SLN> SublineItemDetail { get; }
+        
+        Segment<PID> ItemDescription { get; }
+        
+        Segment<MEA> Measurements { get; }
+        
+        Segment<PKG> MarkingPackagingLoading { get; }
```
Comment (ahives) [inline src/Machete.X12Schema/V5010/Layouts/LoopSLN_846.cs]: should be MarkingPackagingOrLoading

File: src/Machete.X12Schema/V5010/Layouts/LoopSLN_846.cs [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+        
+        LayoutList<LoopMAN_846> LoopMAN_846 { get; }
```
Comment (ahives) [inline src/Machete.X12Schema/V5010/Layouts/LoopSLN_846.cs]: should be LayoutList<LoopMAN_846> LoopMAN { get; }

File: src/Machete.X12Schema/V5010/Layouts/Maps/T846Map.cs [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+
+    public class T846Map :
+        X12LayoutMap<T846, X12Entity>
+    {
+        public T846Map()
+        {
+            Id = "T846";
+            Name = "T846";
+            Segment(x => x.FunctionalGroupHeader, 0);
```
Comment (ahives) [inline src/Machete.X12Schema/V5010/Layouts/Maps/T846Map.cs:14]: should be newline between Name and Segment

File: src/Machete.X12Schema/V5010/Layouts/Maps/LoopSCH_846Map.cs [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+
+    public class LoopSCH_846Map :
+        X12LayoutMap<LoopSCH_846, X12Entity>
+    {
+        public LoopSCH_846Map()
+        {
+            Id = "Loop_SCH_846";
+            Name = "Loop SCH";
+            Segment(x => x.LineItemSchedule, 0);
```
Comment (ahives) [inline src/Machete.X12Schema/V5010/Layouts/Maps/LoopSCH_846Map.cs:15]: should be new line between Name and Segment

File: src/Machete.X12Schema/V5010/Layouts/T846.cs [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        Segment<GS> FunctionalGroupHeader { get; }
+
+        Segment<ST> TransactionSetHeader { get; }
+
+        Segment<BIA> BeginningInventoryInvoice { get; }
+
+        Segment<CUR> Currency { get; }
+
+        SegmentList<DTM> DateTimeReference { get; }
```
Comment (ahives) [inline src/Machete.X12Schema/V5010/Layouts/T846.cs]: Should be DateOrTimeReference

**Review by ahives — CHANGES_REQUESTED** [review]

**Comment by dominiqueplante** [comment]
updated

**Pull request merged**